Assets/Scripts/EOSLobbyManager.cs

```diff
@@ -1590,6 +1592,102 @@ public void ToggleMute(ProductUserId targetUserId, OnLobbyCallback ToggleMuteCom
+        {
+            RTCInterface rtcHandle = EOSManager.Instance.GetEOSRTCInterface();
+            RTCAudioInterface rtcAudioHandle = rtcHandle.GetAudioInterface();
+
+            LobbyMember lobbyMember = GetCurrentLobby().Members.Find(x => x.ProductId == targetProductUserId);
+            if (lobbyMember != null)
+            {
+                // Do not allow multiple local mute toggles at the same time
+                if (lobbyMember.RTCState.MuteActionInProgress)
+                {
+                    Debug.LogWarningFormat("Lobbies (MuteMember): 'MuteActionInProgress' for productUserId {0}.", targetProductUserId);
+                    MuteMemberCompleted?.Invoke(Result.RequestInProgress);
+                    return;
+                }
+
+                // Set mute action as in progress
+                lobbyMember.RTCState.MuteActionInProgress = true;
+            }
+
+            // Check if muting ourselves vs other member
+            if (EOSManager.Instance.GetProductUserId() == targetProductUserId)
+            {
+                // Toggle our mute status
+                UpdateSendingOptions sendOptions = new UpdateSendingOptions()
+                {
+                    LocalUserId = EOSManager.Instance.GetProductUserId(),
+                    RoomName = GetCurrentLobby().RTCRoomName,
+                    AudioStatus = isMicOn ? RTCAudioStatus.Enabled : RTCAudioStatus.Disabled
+                };
+
+                Debug.LogFormat("Lobbies (MuteMember): Setting self audio output status to {0}", sendOptions.AudioStatus == RTCAudioStatus.Enabled ? "Unmuted" : "Muted");
+
+                rtcAudioHandle.UpdateSending(ref sendOptions, MuteMemberCompleted, OnRTCRoomUpdateSendingCompleted);
+            }
+            else
+            {
+                // Toggle mute for remote member (this is a local-only action and does not block the other user from receiving your audio stream)
+
+                UpdateReceivingOptions recevingOptions = new UpdateReceivingOptions()
+                {
+                    LocalUserId = EOSManager.Instance.GetProductUserId(),
+                    RoomName = GetCurrentLobby().RTCRoomName,
+                    ParticipantId = targetProductUserId,
+                    AudioEnabled = isMicOn
+                };
+
+                Debug.LogFormat("Lobbies (MuteMember): {0} remote player {1}", recevingOptions.AudioEnabled ? "Unmuting" : "Muting", targetProductUserId);
+
+                rtcAudioHandle.UpdateReceiving(ref recevingOptions, MuteMemberCompleted, OnRTCRoomUpdateReceivingCompleted);
+            }
+        }
+
+        /// <summary>
+        /// Set the Mute Status of the Local Member
+        /// </summary>
+        /// <param name="micOn">Active status of the Mic</param>
+        /// <param name="MuteLocalMemberCompleted">Callback when MuteLocalMember is completed</param>
+        public void MuteLocalMember(bool micOn, OnLobbyCallback MuteLocalMemberCompleted)
```

Comment:
Perhaps there could be a different name here. It took me a moment to realize that "Local Member" meant "the user that is currently running this". Either we should change the name, or we should add more descriptive comments/documentation.

Reply:
Since we have this, I think any time it's determined that the target of the mute is the local user, we should call this function. Right now, the logic of this function is implemented twice in above functions, I think it should just call this instead.